test/checkout.int.t.sol

```diff
@@ -1119,8 +1171,150 @@ contract Checkout_swapFromSupplierWithoutFeeSpecialOrder is Checkout {
     assertEq(removalAmounts[0], certificateAmount);
     assertEq(
       _bpNori.balanceOf(_namedAccounts.supplier),
-      (certificateAmount * customPriceMultiple) / 100 / 2 // divide to account for price multiple scale and then holdback percentage of 50%
+      (certificateAmount * customPriceMultiple) / 100 / 2 // divide to account for price multiple scale and holdback percentage of 50%
     );
     assertEq(_bpNori.balanceOf(_namedAccounts.feeWallet), 0);
   }
+
+  function test_revertsWhenSupplierDoesNotExistInMarket() external {
+    vm.prank(owner);
+    vm.expectRevert(InsufficientSupply.selector);
+    _market.swapWithoutFeeSpecialOrder({
+      recipient: owner,
+      purchaser: owner,
+      amount: certificateAmount,
+      customFee: customFee,
+      customPriceMultiple: customPriceMultiple,
+      supplier: _namedAccounts.supplier2, // not listed in market
```

Comment:
Given the shared setup function, and the ambiguity of insufficient supply errors (iirc correctly it can also revert with this error if there isn't enough supply vs if there isn't supply) it might be useful to either assert that it doesn't exist in the market first or use a non supplier named account (e.g. namedAccounts.buyer, or _account("non-existent supplier", address(f00)) to prevent accidentally adjusting the shared setup in the future